iothub/service/src/Exceptions/ExceptionHandlingHelper.cs

```diff
@@ -18,181 +16,131 @@ internal class ExceptionHandlingHelper
+            }
+            catch (JsonReaderException ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Info(
+                        nameof(GetIotHubErrorCodeAsync),
+                        $"Failed to parse response content JSON: {ex.Message}. Message body: '{responseBody}.'");
+            }
 
+            if (responseContent != null)
+            {
                 try
                 {
-                    Dictionary<string, string> messageFields = JsonConvert.DeserializeObject<Dictionary<string, string>>(responseContent.Message);
+                    var structuredMessageFields = JsonConvert.DeserializeObject<Dictionary<string, string>>(responseContent.Message);
 
-                    if (messageFields != null
-                        && messageFields.TryGetValue(MessageFieldErrorCode, out string errorCodeObj))
+                    if (structuredMessageFields != null
+                        && structuredMessageFields.TryGetValue(MessageFieldErrorCode, out string errorCodeObj))
                     {
-                        // The result of TryParse is not being tracked since errorCodeValue has already been initialized to a default value of InvalidErrorCode.
-                        _ = int.TryParse(errorCodeObj, NumberStyles.Any, CultureInfo.InvariantCulture, out errorCodeValue);
+                        if (int.TryParse(errorCodeObj, NumberStyles.Any, CultureInfo.InvariantCulture, out int errorCodeInt))
+                        {
+                            return CompareToResponseHeader(response, (IotHubErrorCode)errorCodeInt);
+                        }
                     }
                 }
                 catch (JsonReaderException ex)
                 {
                     if (Logging.IsEnabled)
-                        Logging.Error(null, $"Failed to deserialize error message into a dictionary: {ex}. Message body: '{responseContentStr}.'");
+                        Logging.Info(
+                            nameof(GetIotHubErrorCodeAsync),
+                            $"Failed to deserialize error message into a dictionary: {ex.Message}. Message body: '{responseBody}.'");
+                }
+            }
 
-                    // In some scenarios, the error response string is a ';' delimited string with the service-returned error code.
-                    const char errorFieldsDelimiter = ';';
-                    string[] messageFields = responseContent.Message?.Split(errorFieldsDelimiter);
+            // In some scenarios, the error response string is a semicolon delimited string with the service-returned error code
+            // embedded in a string response.
+            const char errorFieldsDelimiter = ';';
+            string[] messageFields = responseContent.Message?.Split(errorFieldsDelimiter);
 
-                    if (messageFields != null)
+            if (messageFields == null && Logging.IsEnabled)
+            {
+                Logging.Error(
```

Comment:
Now a warning.

Reply:
I just realized that we don't have Warning level logging, do we want to add it if it doesn't break anything?

Reply:
Meh, I guess we'll stick with Error then.